src/openttd.cpp

```diff
@@ -144,6 +144,25 @@ void CDECL error(const char *s, ...)
 	abort();
 }
 
+void CDECL assert_msg_error(int line, const char *file, const char *expr, const char *str, ...)
+{
+	va_list va;
+	char buf[2048];
+
+	char *b = buf;
+	b += seprintf(b, lastof(buf), "Assertion failed at line %i of %s: %s\n\t", line, file, expr);
+
+	va_start(va, str);
+	vseprintf(b, lastof(buf), str, va);
+	va_end(va);
+
+	ShowOSErrorBox(buf, true);
```

Comment:
`error()` does `if (VideoDriver::GetInstance() == nullptr || VideoDriver::GetInstance()->HasGUI()) {`. Possibly good to do that here too?

Otherwise this PR looks good to me!